### DISCUSSION TITLE
Convert old VM or bare-metal to virtink

### DISCUSSION THREAD

**Discussion by Raboo** [smartxworks/virtink#82] [Q&A]
Hi,

Would it be somewhat doable to take some old-legacy-i-never-want-to-touch-this-and-no-one-wants-to-claim-responsibility-for-it virtual machine node running on EOL ubuntu 12.04 and somehow create a virtink vm with persistence?
perhaps rsyncing the old filesystem and copying into a rootfs or create a rootfs from `docker.io/ubuntu:precise` image and rsyncing files over it?

DISCLAIMER I know it's stupid to run end of life stuff and all that crap so please avoid turing this discussion into an besserwisser "the correct way of doing this etc, etc" discussion.

Since hopefully someone will detail the steps I should take or explain somehow that it won't work with ubuntu precise for some reason i feel that I owe a background story perhaps, so here goes.
I do currently work for a company where they don't want me to put time on this and at the same time they don't give the go-ahead for a decommission. But I don't really want this as part of my infrastructure so I would simply like to move this legacy crap to some isolated k8s cluster with virtink running on some cloud and there after give the credentials to my manager and tell him "tag! You're it and no tag-backs!".

**Comment by fengye87** [comment]
@Raboo If you just want to move this VM to Virtink, you can use tools like qemu-img to convert the VM's disk images to QCOW2 format and import it to Virtink. It would require some downtime of the VM, but the VM would be moved to Virtink as it was.

If you don't want any downtime of the VM, some V-to-V technics would be required. It's complicated and should be done with professional tools.

If the node is a bare-metal machine, you could dock its disks to another machine and use qemu-img to convert its disk images to QCOW2 format and import it to Virtink. Again, it would require some downtime for the machine, and if that's unacceptable, some professional P-to-V tools would be required.
